torchft/process_group.py

```diff
@@ -348,6 +350,12 @@ def errored(self) -> Optional[Exception]:
         """
         return None
 
+    def set_timeout(self, timeout: timedelta) -> None:
```

Comment:
this seems to overshadow the existing `_set_default_timeout` in the Backend interface?  i'd be ok with making `_set_default_timeout` public though.  I just don't think we want to have an alternate way of doing the setting

Reply:
Yeah, that's intentional for NCCL since NCCL watchdog often will kill the process.

Reply:
I do agree we should expose set_default_timeout on the PG object directly, current API is subideal and we're depending on private APIs from torchtitan